src/api/ledger/transactions.js

```diff
@@ -1,102 +1,74 @@
-      }
+function hasCompleteLedgerRange(remote, options) {
+  const minLedger = Number(options.minLedger || MIN_LEDGER_VERSION);
+  const maxLedger = Number(options.maxLedger || remote.getLedgerSequence());
+  for (let i = minLedger; i <= maxLedger; i++) {
+    if (!remote.getServer().hasLedger(i)) { // TODO: optimize this
+      return false;
     }
   }
+  return true;
+}
 
-  function queryTransaction(async_callback) {
-    api.remote.requestTx({hash: options.hash}, async_callback);
+function attachTransactionDate(remote, tx, callback) {
+  if (tx.date) {
+    callback(null, tx);
+    return;
   }
-
-  function checkIfRelatedToAccount(transaction, async_callback) {
-    if (options.account) {
-      const transactionString = JSON.stringify(transaction);
-      const account_regex = new RegExp(options.account);
-      if (!account_regex.test(transactionString)) {
-        return async_callback(new errors.InvalidRequestError(
-          'Transaction specified did not affect the given account'));
-      }
-    }
-    async_callback(null, transaction);
+  if (!tx.ledger_index) {
+    callback(new errors.ApiError('ledger_index not found in tx'));
+    return;
   }
 
-  function attachDate(transaction, async_callback) {
-    if (!transaction || transaction.date || !transaction.ledger_index) {
-      return async_callback(null, transaction);
+  remote.requestLedger(tx.ledger_index, (error, data) => {
+    if (error) {
+      callback(new errors.NotFoundError('Transaction ledger not found'));
+    } else if (typeof data.ledger.close_time === 'number') {
+      callback(null, _.assign({date: data.ledger.close_time, tx}));
+    } else {
+      callback(new errors.ApiError('Ledger missing close_time'));
     }
+  });
+}
 
-    api.remote.requestLedger(transaction.ledger_index,
-        function(error, ledgerRequest) {
-      if (error) {
-        return async_callback(new errors.NotFoundError(
-          'Transaction ledger not found'));
-      }
+function isTransactionInRange(tx, options) {
+  return (!options.minLedger || tx.ledger_index >= options.minLedger)
+      && (!options.maxLedger || tx.ledger_index <= options.maxLedger);
+}
 
-      if (typeof ledgerRequest.ledger.close_time === 'number') {
-        transaction.date = ledgerRequest.ledger.close_time;
-      }
+function getTransaction(identifier, options, callback) {
+  validate.identifier(identifier);
+  validate.options(options);
 
-      async_callback(null, transaction);
-    });
-  }
+  const remote = this.remote;
 
-  const steps = [
-    queryTransaction,
-    checkIfRelatedToAccount,
-    attachDate
-  ];
+  function callbackWrapper(error, tx) {
+    if (error instanceof errors.NotFoundError
+        && !hasCompleteLedgerRange(remote, options)) {
+      callback(new errors.MissingLedgerHistoryError('Transaction not found,'
+        + ' but the server\'s ledger history is incomplete'));
+    } else if (!error && !isTransactionInRange(tx, options)) {
```

Comment:
There's not much need not to expose it. That case explicitly catches the
case where it was found but out of range.  You can still throw an error,
just be more precise, with the reporting, so they get some inkling of
what's going on.

Much like, "but the ledger history is incomplete"